File: aegis-platform-web/src/types/tenant.ts
/**
 * @file 租户类型定义
 * @description 租户、租户配额、租户用量、租户上下文相关类型，对齐后端实体
 * @author wang.zhen
 * @since 1.0.0
 */

/** 租户信息 */
export interface Tenant {
  /** 租户 ID */
  id: string;
  /** 租户编码（唯一） */
  tenantCode: string;
  /** 租户名称 */
  tenantName: string;
  /** 租户类型：HQ 总部 / SUBSIDIARY 分公司 / DIVISION 事业部 */
  tenantType: string;
  /** 租户状态：NORMAL 正常 / FROZEN 冻结 */
  status: string;
  /** 联系人 */
  contactName?: string;
  /** 联系电话 */
  contactPhone?: string;
  /** 到期时间 */
  expireTime?: string;
  /** 备注 */
  remark?: string;
  /** 创建时间 */
  createTime?: string;
}

/** 租户配额 */
export interface TenantQuota {
  /** 配额 ID */
  id?: string;
  /** 租户 ID */
  tenantId?: string;
  /** 智能体数量上限 */
  maxAgents: number;
  /** 资源数量上限 */
  maxResources: number;
  /** 并发会话上限 */
  maxConcurrentSessions: number;
  /** 每日 Token 上限 */
  maxTokenPerDay: number;
  /** 每月 Token 上限 */
  maxTokenPerMonth: number;
  /** 沙箱数量上限 */
  maxSandboxes: number;
  /** 存储上限（GB） */
  maxStorageGb: number;
}

/** 租户用量（实时统计） */
export interface TenantUsage {
  /** 租户 ID */
  tenantId?: string;
  /** 当前智能体数 */
  agentCount: number;
  /** 当前资源数 */
  resourceCount: number;
  /** 当前并发会话数 */
  concurrentSessionCount: number;
  /** 今日已用 Token */
  tokenUsedToday: number;
  /** 本月已用 Token */
  tokenUsedThisMonth: number;
  /** 已用沙箱数 */
  sandboxUsed: number;
  /** 已用存储（GB） */
  storageUsedGb: number;
  /** 统计日期 */
  statDate?: string;
}

/** 租户上下文（贯穿全链路） */
export interface TenantContext {
  /** 租户 ID */
  tenantId: string;
  /** 租户编码 */
  tenantCode: string;
  /** 租户名称 */
  tenantName: string;
}

File: aegis-platform-backend/aegis-core/aegis-core-domain/src/main/java/com/aegis/core/domain/tenant/Tenant.java
package com.aegis.core.domain.tenant;

import com.aegis.core.base.BaseEntity;
import com.aegis.core.enums.tenant.TenantStatus;
import com.aegis.core.enums.tenant.TenantType;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;
import com.aegis.core.base.TenantEntity;

/**
 * 租户实体，平台多租户隔离的根聚合。
 *
 * <p>对应平台多租户体系的根聚合根，承载租户的基础属性、联系人、有效期与状态。
 * 继承 BaseEntity（无 tenantId，因为 Tenant 本身即租户主体）。
 * 所有租户隔离实体通过 tenantId 关联本实体实现数据隔离。
 *
 * <h3>设计约束</h3>
 * <ul>
 *   <li>租户编码（tenantCode）全局唯一，创建后不可修改</li>
 *   <li>租户状态决定其下所有资源可用性：0-停用 1-正常</li>
 *   <li>过期租户（expireTime 早于当前时间）自动停用，拒绝新建会话</li>
 * </ul>
 *
 * <h3>关联实体</h3>
 * <ul>
 *   <li>{@link TenantQuota} - 租户配额配置</li>
 *   <li>{@link TenantUsage} - 租户用量统计</li>
 *   <li>{@link com.aegis.core.base.TenantEntity} - 租户隔离实体基类</li>
 * </ul>
 *
 * @author wang.zhen
 * @see TenantQuota
 * @see TenantUsage
 */
@Getter
@Setter
@ToString
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@TableName("ten_tenant")
public class Tenant extends BaseEntity {

    /** 租户编码，全局唯一，业务可读标识，用于URL与配置引用，创建后不可修改 */
    private String tenantCode;

    /** 租户名称，展示用，可重复，支持修改 */
    private String tenantName;

    /** 租户类型：{@link TenantType#HQ}（集团总部）、{@link TenantType#SUBSIDIARY}（子公司）、{@link TenantType#DIVISION}（事业部），决定默认配额档位 */
    private TenantType tenantType;

    /** 租户状态：{@link TenantStatus#NORMAL}（正常）、{@link TenantStatus#FROZEN}（冻结），冻结后拒绝新建会话与资源操作 */
    private TenantStatus status;

    /** 租户联系人姓名，用于运营对接 */
    private String contactName;

    /** 租户联系人电话，用于运营对接与紧急通知 */
    private String contactPhone;

    /** 租户有效期截止时间，过期自动停用，null 表示长期有效 */
    private LocalDateTime expireTime;

    /** 备注，运营自定义描述信息 */
    private String remark;
}
